### ISSUE TITLE
truffle workflow-compile lost dashboard-message-bus-client dependency

### ISSUE THREAD

**Issue by moshmage** [trufflesuite/truffle#5235]
## Issue

The package lost dependencies but still relies on them

## Steps to Reproduce

Visit this public github action log, check build stage.
[Please provide the shortest amount of steps to reproduce your issue.](https://github.com/taikai/dappkit/runs/7069036809?check_suite_focus=true)

## Expected Behavior

The compile action to work

## Actual Results

truffle/workflow-compile complains about missing dependency,
```
Error: Cannot find module '@truffle/dashboard-message-bus-client'
```

## Environment

* Operating System: 
* Ethereum client:
* Truffle version: "@truffle/workflow-compile": "4.0.20",
* node version:  v14.17.0
* npm version: 6.14.13

**Comment by moshmage** [comment]
after providing `dashboard-message-bus-client` manually via package.json dependencies, this shows up

```
internal/modules/cjs/loader.js:885
  const err = new Error(message);
              ^
Error: Cannot find module './messages'
Require stack:
- /home/runner/work/dappkit/dappkit/node_modules/@truffle/dashboard-message-bus-common/dist/index.js
- /home/runner/work/dappkit/dappkit/node_modules/@truffle/dashboard-message-bus-client/dist/client.js
- /home/runner/work/dappkit/dappkit/node_modules/@truffle/dashboard-message-bus-client/dist/index.js
- /home/runner/work/dappkit/dappkit/node_modules/@truffle/events/defaultSubscribers/dashboard.js
- /home/runner/work/dappkit/dappkit/node_modules/@truffle/events/defaultSubscribers/index.js
- /home/runner/work/dappkit/dappkit/node_modules/@truffle/events/EventManager.js
- /home/runner/work/dappkit/dappkit/node_modules/@truffle/events/index.js
- /home/runner/work/dappkit/dappkit/node_modules/@truffle/config/dist/index.js
- /home/runner/work/dappkit/dappkit/node_modules/@truffle/workflow-compile/utils.js
- /home/runner/work/dappkit/dappkit/node_modules/@truffle/workflow-compile/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:885:15)
```

Something _very wrong_ happened on your last deploy - This is essentially blocking us from deploying a new version.

**Comment by cds-amal** [comment]
Hi @moshmage, thanks for raising this dependency issue. Hopefully #5230 resolves this.

**Comment by moshmage** [comment]
> Hey @moshmage, [the fix](https://github.com/trufflesuite/truffle/pull/5246) should make it into this week's release. Thanks for providing a workaround by pinning the version!

Hey @cliffoo while that fix _might_ solve some instances on the truffle plugin itself, if changes aren't made to the dependencies on what truffle depends on we will keep having these problems as every dependency on every package on truffle repository uses a "at least" approach, meaning that whenever you bump to `0.0.2` and introduce new stuff, even if I require `0.0.1` _the dependencies_ will still require `0.0.2` because the dependencies on `0.0.1` are written as `^0.0.1` which will target `0.0.2`

case in point [This dependency will not let me _choose_ a version, it will always install the latest on on that version](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle/package.json#L46)

And since clearly we are not respecting semver (or the changes to events would have raised a version instead of a _patch_) we need to lock each version to its deployed counterpart _or this will keep happening_

**Comment by cliffoo** [comment]
@moshmage It's not that semver isn't respected, but a lapse on our end that caused this breaking behavior to slide into the release. We probably don't want to pin versions as there are other implications.

We are looking to integrate CI checks (outlined in [this PR](https://github.com/trufflesuite/truffle/pull/5246)) that can make us aware of such issues (by simulating install from registry) and gives us a chance to fix them before merging to develop.

**Comment by lsqproduction** [comment]
Yay. Fixed in [Truffle 5.5.20](https://github.com/trufflesuite/truffle/releases/tag/v5.5.20)
